lib/start-command.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+var debug = require('./debug')('start-command');
+var fs = require('fs');
+var path = require('path');
+
+module.exports = exports = resolveArgs;
+
+// for tests
+exports.fromPath = fromPath;
+exports.resolveFile = resolveFile;
+exports.resolveDir = resolveDir;
+exports.fromStart = fromStart;
+
+function resolveArgs(cwd, argv) {
+  var script = argv[2] || '.';
+  return fromPath(cwd, script);
+}
+
+function fromPath(cwd, script) {
+  var app = {
+    cwd: cwd,
+    path: script,
+    error: null,
+  };
+
+  try {
+    var stat = fs.statSync(path.resolve(app.cwd, app.path));
+  } catch (er) {
+    try {
+      app.path = require.resolve(path.resolve(app.cwd, app.path));
+      stat = fs.statSync(app.path);
+    } catch (er) {
+      app.error = er;
+      return app;
+    }
+  }
+
+  // We want to be in the directory of the file we are running, so we can pick
+  // up configuration stored in it's working directory.
+
+  if (stat.isFile()) {
+    // The app is a file, we'll run it from its module's root directory.
+    return resolveFile(cwd, script);
+  } else if (stat.isDirectory()) {
+    // The app is a directory (such as '.'), we'll look for what to run from it.
+    try {
+      require.resolve(path.resolve(cwd, script));
+      return resolveDir(cwd, script);
+    } catch (err) {
+      // given path is not resolvable as a module
+      return resolvePackageStart(cwd, script);
```

Comment:
so, we only look at package.start if package has neither and index.js, nor a package.main? I think package.start should  be more definitive. If you `slc run a-file-that-exists`, then run that file, if you slc run a package, then the order of resolution should be:
- server.js
- app.js
- package.main
- index.js

btw, this PR changes how apps are resolved, but doesn't touch bin/sl-run.txt, and its docs on resolution.